src/pages/Vault/index.tsx

```diff
@@ -93,15 +108,20 @@ const Vault = (): JSX.Element => {
     setRequestReplacementModalOpen(true);
   };
 
+  const vaultAccountId =
```

Comment:
No variables directly in the functions please. This has a very bad impact on performance and causes a ton of rerenders.

you can check that with this change, my vault dashboard can't load anymore since it's taking forever to load: https://interbtc-ui-testnet-git-fork-crypto-engineer-fe-90ace3-interlay.vercel.app/vault/5HTQFN3xWmEPFiXRtFiuWDWWKjoV8gkUWNqUT57tZChvvKFV

Reply:
@nud3l 
I improved performance of this vault page. But the current structure of this component is not good. I just improved the performance in consideration of time and priority. Actually, we should optimize this page by better componentizing rather than using quick fixes or workarounds like `useMemo` for scaling and maintaining. Just for your information for now.

The underlying problem was that `window.bridge.api.createType` is a computationally expensive to calculate. There are (too) many state updates in this component which cause re-renders because this component has many `useState` and `dispatch` used in a non-optimal (actually ruthless) way.
Every time this component re-renders, `window.bridge.api.createType` used to be called which causes a long task, which causes the browser's JavaScript runtime to be blocked, which causes unresponsiveness to the user's interactions with the UI.

I fixed this by memoizing this computationally expensive calculation but I'd like to suggest revisiting this component to refactor the componentization properly.
Could you please test it out on your side and confirm that this has been fixed? Thank you.

Reply:
@nud3l cc @savudani8 
I'd like us to know what methods are computationally expensive calculations so that we (front-end) can handle them pro-actively.

Reply:
There are two issues here:
1. The performance of this page is not good. We will rebuild it based on the designs that are currently at work and better componentize it. This will also include an overhaul of the UX to make it better for vaults overall. I'm working on making sure that we get these designs ready soon and can work on it.
2. The PR was undertested and therefore the code quality was lacking as it decreased the already suboptimal performance further. I need you to work on that part. While the page is not optimal, it is on a quality level not acceptable that only if I review this PR that performance issues become apparent. As a contributor, it is your responsibility to ensure that performance is kept or improve where necessary.

Reply:
> @nud3l cc @savudani8 I'd like us to know what methods are computationally expensive calculations so that we (front-end) can handle them pro-actively.

Any `createType` operation is very expensive. Any transaction takes a very long time to calculate. Any query to the blockchain directly (i.e., any queries via the lib) are slow (1000ms or longer).

Reply:
@nud3l 
> Any transaction takes a very long time to calculate. Any query to the blockchain directly (i.e., any queries via the lib) are slow (1000ms or longer).

Promises are very fine because they are not long tasks. Expensive sync functions like `createType`, hashing had better be wrapped in `useMemo` in that case. I'll do so from now on.

Reply:
@nud3l 
> The PR was undertested and therefore the code quality was lacking as it decreased the already suboptimal performance further. I need you to work on that part. While the page is not optimal, it is on a quality level not acceptable that only if I review this PR that performance issues become apparent. As a contributor, it is your responsibility to ensure that performance is kept or improve where necessary.

I already optimized the page and it had no problem on my side. Is it still problematic on your side?

Reply:
Working now, thanks!